File: Data Structures & Algorithms/duplicate-integer/submission-0.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        set<int> s;
        for (auto a : nums) {
            if (s.count(a) > 0) {
                return true;
            }
            s.insert(a);
        }
        return false;
    }
};

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.ts
class Solution {
    /**
     * @param {number[]} nums
     * @return {number[]}
     */
    productExceptSelf(nums: number[]): number[] {
        let l = new Array(nums.length + 1).fill(1),
            r = new Array(nums.length + 1).fill(1);
        for (let i = 0; i < nums.length; i++) {
            l[i + 1] = l[i] * nums[i];
            r[nums.length - i - 1] = r[nums.length - i] * nums[nums.length - i - 1]; 
        }

        let ans = new Array(nums.length);
        for (let i = 0; i < nums.length; i++) {
            ans[i] = l[i] * r[i+1];
        }
        return ans;
    }
}
